### PR TITLE
feat(indexbar): 增加索引组件

### PR DESCRIPTION
**Pull request by MNZhu**
# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/ui-vue/blob/main/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/index-bar-anchor/vue.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ * Use of this source code is governed by an MIT-style license.
+ *
+ * THE OPEN SOURCE SOFTWARE IN THIS PRODUCT IS DISTRIBUTED IN THE HOPE THAT IT WILL BE USEFUL,
+ * BUT WITHOUT ANY WARRANTY, WITHOUT EVEN THE IMPLIED WARRANTY OF MERCHANTABILITY OR FITNESS FOR
+ * A PARTICULAR PURPOSE. SEE THE APPLICABLE LICENSES FOR MORE DETAILS.
+ *
+ */
+
+import {} from './index'
```
Comment (zzcr) [inline src/index-bar-anchor/vue.js:13]: 如果是空文件可以删除，后期如果需要再添加

File: src/index-bar/vue.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    handleTouchUp: handleTouchUp({ state }),
+    handleIndexClick: handleIndexClick({ emit, parent, refs, state })
+  }
+
+  Object.assign(api, {
+    state
+  })
+
+  onMounted((e) => {
+    window.addEventListener('scroll', api.handleScroll)
```
Comment (zzcr) [inline src/index-bar/vue.js:49]: 这里在组件卸载的时候，需要取消监听

File: src/index-bar-anchor/vue.js (hunk 1 above)
Comment (MNZhu) [reply src/index-bar-anchor/vue.js:13]: 已删除

File: src/index-bar/vue.js (hunk 2 above)
Comment (MNZhu) [reply src/index-bar/vue.js:49]: 已增加取消监听逻辑

**Review by zzcr — APPROVED** [review]

**Pull request merged**